pkgs/applications/terminal-emulators/oscilloscope-terminal/default.nix

```diff
@@ -0,0 +1,50 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, SDL2
+, bash
+, shell ? bash # dunno how to override this...
+, WIDTH ? 80
+, HEIGHT ? 24
+, BLOCK_SIZE ? 2048
+, BLOCKSFRAME ? 4
+, SAMP_RATE ? 192000
+}:
+
+stdenv.mkDerivation rec {
+  pname = "oscilloscope-terminal";
+  version = "unstable-2023-05-12";
+
+  src = fetchFromGitHub {
+    owner = "arf20";
+    repo = "oscilloscope-terminal";
+    rev = "dad28ec3c4e8c75d1400f387370491f4c868dda9";
+    hash = "sha256-fjM1g6YM6+Sen4/rzNXduuDDVD2eSV+UdDPzJUvW0SY=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [ SDL2 ];
+
+  postPatch = ''
+    substituteInPlace CMakeLists.txt \
+      --replace "target_link_libraries(oscilloscope-terminal Threads::Threads SDL2main SDL2 GL)" \
+                "target_link_libraries(oscilloscope-terminal Threads::Threads SDL2 GL)"
+
+    substituteInPlace terminal.cpp --replace "/bin/bash" "${shell}${shell.shellPath}"
+
+    substituteInPlace main.hpp --replace "WIDTH   80" "WIDTH   ${toString WIDTH}"
```

Comment:
It'd be nice to configure these with environment variables. It'd be nice to suggest this to upstream.